File: 485-max-consecutive-ones/max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0,ans(0);
       for (int x : nums){
         if (x){
            count++;
         }
         else {
           ans= max(count,ans);
          
            count=0;
         }
       }
       return   max(count,ans);;
    }
};

File: 485-max-consecutive-ones/max-consecutive-ones.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let ans=0;
    let sum=0;
    for(const i of nums){
        if(i==0){
            sum=0;
        }
        else{
            sum++;
            ans=Math.max(ans,sum);
        }
    }
    return ans;
};